### PR TITLE
Memento Property states documentation

### PR DESCRIPTION
**Pull request by mauroservienti**
Connects to https://github.com/RadicalFx/Radical/issues/236

@nazarenomanco please review

### CODE REVIEW

**Comment by mauroservienti** [comment]
Good point @nazarenomanco I'll review it. Probably it's much better to split this `API` article into sub-articles extracting concepts around entity and property states.

**Comment by mauroservienti** [comment]
@nazarenomanco what about now?

**Comment by mauroservienti** [comment]
@nazarenomanco ping 😄

**Comment by mauroservienti** [comment]
I'll merge this as soon as `Radical 1.7.0` is released.

File: docs/memento/change-tracking-service-api.md [hunk 1]
```diff
@@ -6,22 +6,21 @@ All the memento entities implement the `IMemento` interface.
...
-* `HasTransientEntities` determines if the the memento is currently tracking transient entities;
-* `GetEntityState( Object entity )` return, given a tracked entity, the current entity state as seen by the memento, the returned value is a `EntityTrackingStates` enumeration that can assume one, or more, of the following values:
+*   `Attach( IMemento item )` / `Detach( IMemento entity )`: attach and detach are the 2 methods to manually control when an instance is attached or detached to and from the memento instance. As soon as an instance is attached it will be tracked for changes and the memento will stop tracking it at detach time.
+*   `Undo()` /  `Redo()`: Undo and Redo controls the state of the tracked entities, calling `Undo` will revert the last tracked operation, if any, calling `Redo` will apply the last operation that has been undone, if any;
+*   `CanRedo` and `CanUndo` allows the user code to determine if calling Undo and Redo operations something will be done; 
+*   `RegisterTransient( Object entity )` and `RegisterTransient( Object entity, Boolean autoRemove )` allows the user to register an entity as a transient, versus persistent, entity.
```
Comment (nazarenomanco) [inline docs/memento/change-tracking-service-api.md]: Do we have to clarify better the difference between transient and persistent entity in this scenario?

File: docs/memento/change-tracking-service-api.md [hunk 2]
```diff
@@ -6,22 +6,21 @@ All the memento entities implement the `IMemento` interface.
...
+*   `Undo()` /  `Redo()`: Undo and Redo controls the state of the tracked entities, calling `Undo` will revert the last tracked operation, if any, calling `Redo` will apply the last operation that has been undone, if any;
+*   `CanRedo` and `CanUndo` allows the user code to determine if calling Undo and Redo operations something will be done; 
+*   `RegisterTransient( Object entity )` and `RegisterTransient( Object entity, Boolean autoRemove )` allows the user to register an entity as a transient, versus persistent, entity.  
+Registering transient entities is not really required for the memento to work properly, it is on the other hand very handy for the user if the code needs at a certain point to deal with a storage trying to understand which operations should be done to align the storage with the current in memory state. if `autoRemove` is set to `true` (*the default value*) and `RejectChanges()`, or an `Undo()` that removes the last `IChange` of the object, is called the object then is automatically removed from the list of the new objects.
```
Comment (nazarenomanco) [inline docs/memento/change-tracking-service-api.md]: Sorry, but for me it is not so clear this "note".

File: docs/memento/change-tracking-service-api.md [hunk 3]
```diff
@@ -6,22 +6,21 @@ All the memento entities implement the `IMemento` interface.
...
+*   `UnregisterTransient( Object entity )` manually remove the given transient entity from the list of transient entities;
+*   `HasTransientEntities` determines if the the memento is currently tracking transient entities;
+*   `GetEntityState( Object entity )` return, given a tracked entity, the current entity state as seen by the memento, the returned value is a `EntityTrackingStates` enumeration that can assume one, or more, of the following values:
     *  `None`:  The state of the entity is not changed, the entity is not transient or the entity is not tracked;
     *  `IsTransient`: The entity is registered as transient;
     *  `AutoRemove`: if an entity is marked as `AutoRemove` (the default behavior) and `RejectChanges`, or an `Undo` that removes the last `IChange` of the entity, is called then the entity is  automatically removed from the list of the transient entities;
```
Comment (nazarenomanco) [inline docs/memento/change-tracking-service-api.md]: If it is a property... "IsRemoved" can be a better "name" of the property?

Comment (mauroservienti) [reply docs/memento/change-tracking-service-api.md]: `AutoRemove` is not a property, it’s one of the values of the `EntityTrackingStates` enumeration. And it’s not related to properties. In this context we’re not going to change it, hence I’ll resolve this comment with no further changes.

File: docs/memento/change-tracking-service-api.md (hunk 2 above)
Comment (mauroservienti) [reply docs/memento/change-tracking-service-api.md]: I’ll reword the whole paragraph in a different PR. Good point.

File: docs/memento/change-tracking-service-api.md (hunk 1 above)
Comment (mauroservienti) [reply docs/memento/change-tracking-service-api.md]: Yes, I’ll do that in a separate PR: https://github.com/RadicalFx/documentation/issues/67

File: docs/memento/change-tracking-service-api.md (hunk 2 above)
Comment (mauroservienti) [reply docs/memento/change-tracking-service-api.md]: https://github.com/RadicalFx/documentation/issues/67

**Comment by mauroservienti** [comment]
@nazarenomanco I addressed all your comments, if the content of the `property-state.md` file looks good, can you please approve the PR?

**Pull request merged**